src/lazy/text/buffer.rs

```diff
@@ -2817,6 +2899,31 @@ mod tests {
         assert!(contains_escapes);
     }
 
+    #[rstest]
+    #[case::no_whitespace("", (1,1))]
+    #[case::whitespace_without_newlines("\t\t", (1,3))]
+    #[case::newlines("\n\r", (3,1))]
+    #[case::crlf("\r\n\r\n", (3,1))]
+    #[case::mixed("\r\n\n\r\n", (4,1))]
+    #[case::tabs("\n\t\t\t", (2,4))]
+    #[case::mix_tabs_and_newlines("\n\t\n", (3,1))]
+    fn expect_whitespace(#[case] input: &str, #[case] expected_location: (usize, usize)) {
+        MatchTest::new_1_0(input).expect_match_location(match_length(TextBuffer::match_whitespace0), expected_location);
+    }
+
+    #[rstest]
+    #[case::no_whitespace("", (1,1))]
+    #[case::whitespace_with_inline_comment("//comment ", (1,11))]
+    #[case::whitespace_with_comment("/*comment*/ \n", (2,1))]
+    fn expect_whitespace_with_comment(#[case] input: &str, #[case] expected_location: (usize, usize)) {
+        MatchTest::new_1_0(input).expect_match_location(match_length(TextBuffer::match_optional_comments_and_whitespace), expected_location);
+    }
+
+    #[test]
+    fn expect_newline_long_text() {
```

Comment:
Suggested test cases
* Empty single-segment long string.
* Single segment long string containing only `\n`
* Single segment long string containing other stuff
* Two-segment long string on a single line (i.e. `'''a''' '''b'''`)
* Two-segment long string on separate lines (i.e. `'''a'''\n'''b'''`)
* Two-segment long string containing new lines (i.e. `'''a\n''' '''b\n'''`)
* Two-segment long string containing new lines and on separate lines (i.e. `'''a\n'''\n'''b\n'''`)

Also, I think it's probably worth adding (or generalizing this to include) some basic test cases for blobs, clobs, list, sexp, and structs that contain newlines; newlines between annotations; and newlines between field names and field values.